core/src/main/java/com/graphhopper/routing/util/parsers/OSMTollParser.java

```diff
@@ -49,10 +47,74 @@ public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay rea
         } else {
             toll = Toll.MISSING;
         }
-        
-        CountryRule countryRule = readerWay.getTag("country_rule", null);
-        if (countryRule != null)
-            toll = countryRule.getToll(readerWay, toll);
+
+        if (toll == Toll.MISSING) {
+            Country country = readerWay.getTag("country", Country.MISSING);
+            switch (country) {
```

Comment:
Can we move this to a dedicated method?

```java
    Country country = readerWay.getTag("country", Country.MISSING);
    toll = applyCountryDefaults(country, toll);
```

This keeps the `handleWayTags` short and allows to get rid of all the `break` statements by using `return` instead.